pkg/sources/kafka/reader.go

```diff
@@ -391,7 +394,8 @@ func (r *KafkaSource) startConsumer() {
 			// server-side rebalance happens, the consumer session will need to be
 			// recreated to get the new claims
 			if err := client.Consume(r.lifecyclectx, []string{r.topic}, r.handler); err != nil {
-				r.logger.Warnw("Initialization of consumer failed with error: ", zap.Error(err))
+				// Panic on errors to let it crash and restart the process
+				r.logger.Panicw("Consumer failed with error: ", zap.Error(err))
```

Comment:
Reference to https://github.com/Shopify/sarama/blob/main/examples/consumergroup/main.go#L109.

Reply:
Does this return error and cause panic due to repartitioning on the server side? If yes, we shouldn't panic. We should do a manual repartition on a local kafka cluster and try it out.

I skimmed through the Sarama code, but it wasn't quite obvious to me.

Reply:
> Does this return error and cause panic due to repartitioning on the server side? If yes, we shouldn't panic. We should do a manual repartition on a local kafka cluster and try it out.
> 
> I skimmed through the Sarama code, but it wasn't quite obvious to me.

I didn't find any `Initialization of consumer failed with error` logs in our prod pipelines in the last 15 days, I assume repartitioning won't cause an error.